### PR TITLE
166 homepage

### PR DESCRIPTION
**Pull request by agentrickard**
This is close to working, and based on the approach in #166.

However, URL paths on the generated page are inconsistent. I am running in a subdirectory, and the base_path() is being ignored, breaking the page output.

PR opened so people can look at the code.

See also https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php#L56

### CODE REVIEW

**Comment by bjaxelsen** [comment]
Works fine on my setup, thanks!

However, I can't make the `DomainConfigHomepageTest::testDomainConfigOverrider()` test pass.

The "Proper home page loaded." test fails. This is due to this error:

```
Undefined index: four_localhost
Drupal\domain_config\Tests\DomainConfigHomepageTest->testDomainConfigOverrider()
```

[test.html.zip](https://github.com/agentrickard/domain/files/112673/test.html.zip)

File: domain_config/src/Tests/DomainConfigHomepageTest.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      $expected = $domain->getPath() . $homepages[$domain->id()];
+      $expected_home = $this->drupalGet($expected);
+      $this->assertTrue($home == $expected_home, 'Proper home page loaded.');
+    }
+  }
+
+  /**
+   * Returns the expected homepage paths for each domain.
+   */
+  private function getHomepages() {
```
Comment (agentrickard) [inline domain_config/src/Tests/DomainConfigHomepageTest.php]: Rewrite this part to use the keys from actual domains.

**Comment by agentrickard** [comment]
The logic in getHomepages() hardcodes machine names. That needs to be changed.

**Comment by agentrickard** [comment]
I suspect the failures that I get are because I am running in a subdirectory. That worries me, because it means that the code isn't working in the same way as core.

Did you see any issues with active link highlighting or breadcrumbs?

**Comment by bjaxelsen** [comment]
In the test result? No, I don't think so (like a written breadcrumb?), have a look here:
[test_results.docx](https://github.com/agentrickard/domain/files/112749/test_results.docx)

I haven't tried running in a subpath.

**Comment by agentrickard** [comment]
I had problems with breadcrumbs and links when using the UI, not in tests.

For tests, I suspect that none of the Domain Config tests will pass for you, since the schema files (the config overrides) are keyed to example_com not localhost. We could fix that by changing this part of the test domain creation routine:

```
      // Create a new domain programmatically.
      $values = array(
        'hostname' => $hostname,
        'name' => $name,
        'id' => \Drupal::service('domain.creator')->createMachineName($hostname),
      );
```

We should use example_com (or localhost) consistently. domainCreateTestDomains() handles different hosts properly (otherwise none of the tests would pass), but it is mismatching on the machine name of the domain.

**Comment by agentrickard** [comment]
Pushed a commit that should help with that specific test fail.

**Pull request closed without merging**